File: C#/T1/Project/W5_Exercise2/Pet.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace W5_Exercise2
{
    class Pet
    {
        private string Name;
        private float Hunger = 100.0f;
        private float Thirst = 100.0f;
        private float Sleepiness = 100.0f;
        private float Happiness = 100.0f;

        public Pet()
        {
            Name = "DEFAULT NAME";
        }

        public Pet(string pName)
        {
            Name = pName;
        }

        public void Eat(float amount)
        {
            Hunger -= amount;
            if(Hunger < 0)
            {
                Hunger = 0;
            }
        }

        public void RollOver(float pHunger, float pSleep, float pHappy)
        {
            Hunger += pHunger;
            Sleepiness += pSleep;
            Happiness += pHappy;

            if(Hunger > 100.0f)
            {
                Hunger = 100.0f;
            }

            if (Sleepiness > 100.0f)
            {
                Sleepiness = 100.0f;
            }

            if (Happiness > 100.0f)
            {
                Happiness = 100.0f;
            }
        }

        public void setName(string pName)
        {
            Name = pName;
        }

        public void setHunger(float pHunger)
        {
            Hunger = pHunger;
        }

        public void setThirst(float pThirst)
        {
            Thirst = pThirst;
        }

        public void setSleepiness(float pSleep)
        {
            Sleepiness = pSleep;
        }

        public void setHappiness(float pHappy)
        {
            Happiness = pHappy;
        }

        public string getName()
        {
            return Name;
        }

        public float getHunger()
        {
            return Hunger;
        }

        public float getThirst()
        {
            return Thirst;
        }

        public float getSleepiness()
        {
            return Sleepiness;
        }

        public float getHappiness()
        {
            return Happiness;
        }
    }
}

File: C++/T2/PetSimulator/PetSimulator/Pet.cpp
#include <iostream>

#include "ConsoleColor.h"
#include "Pet.h"

using namespace std;

ConsoleColor cc;

Pet::Pet() :
	petName("DEFAULT NAME"),
	hunger(10.0f),
	thirst(10.0f),
	sleepiness(10.0f),
	happiness(10.0f)
{}

Pet::~Pet()
{}

void Pet::setName(std::string name)
{
	petName = name;
}

std::string Pet::getName()
{
	return petName;
}

float Pet::getHunger()
{
	return hunger;
}

float Pet::getThirst()
{
	return thirst;
}

float Pet::getSleepiness()
{
	return sleepiness;
}

float Pet::getHappiness()
{
	return happiness;
}

void Pet::Eat(float amount)
{
	hunger += amount;

	checkStats();
}

void Pet::Walk(float distance)
{
	hunger -= (distance / 80);
	thirst -= (distance / 50);
	sleepiness -= (distance / 80);
	happiness -= (distance / 100);

	checkStats();
}

void Pet::Sleep(float amount)
{
	hunger -= (amount * 5);
	thirst -= (amount * 3);
	sleepiness += (amount * 10);
	happiness -= (amount * 10);

	checkStats();
}

void Pet::Playing(float amount)
{
	hunger -= (amount * 8);
	thirst -= (amount * 5);
	sleepiness -= (amount * 6);
	happiness += (amount * 5);

	checkStats();
}

void Pet::WatchMovie(float amount)
{
	hunger -= (amount * 2);
	thirst -= (amount * 2);
	sleepiness -= (amount * 3);
	happiness += (amount * 10);

	checkStats();
}

void Pet::Toilet()
{
	hunger -= 5;
	thirst -= 5;
	sleepiness -= 5;
	happiness += 5;

	checkStats();
}

bool Pet::checkHealth()
{
	if (hunger == 0.0f || thirst == 0.0f || sleepiness == 0.0f || happiness == 0.0f)
	{
		return false;
	}
	else if (hunger < 10.0f || thirst < 10.0f || sleepiness < 10.0f || happiness < 10.0f)
	{
		cc.Magenta();
		cout << endl << "[" << petName << "]";
		cc.White();
		cout << " I'm dying" << endl;

		return true;
	}
	else
	{
		return true;
	}
}

void Pet::Reset()
{
	hunger = 100.0f;
	thirst = 100.0f;
	sleepiness = 100.0f;
	happiness = 100.0f;
}

void Pet::checkStats()
{
	if (hunger > 100.0f)
	{
		hunger = 100.0f;
	}

	if (hunger < 0.0f)
	{
		hunger = 0.0f;
	}

	if (thirst > 100.0f)
	{
		thirst = 100.0f;
	}

	if (thirst < 0.0f)
	{
		thirst = 0.0f;
	}

	if (sleepiness > 100.0f)
	{
		sleepiness = 100.0f;
	}

	if (sleepiness < 0.0f)
	{
		sleepiness = 0.0f;
	}

	if (happiness > 100.0f)
	{
		happiness = 100.0f;
	}

	if (happiness < 0.0f)
	{
		happiness = 0.0f;
	}
}
